generators/pkg/generator/generate.go

```diff
@@ -25,3 +45,94 @@ func GenerateTask(spec GitHubSpec) v1beta1.Task {
 	)
 	return task
 }
+
+func addTask(url string) (v1beta1.Task, error) {
+	task := v1beta1.Task{}
+	response, err := http.Get(url)
+	if err != nil {
+		return task, fmt.Errorf("fail to get git-clone task from %s: %w", url, err)
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return task, fmt.Errorf("fail to read reponse body: %v", err)
```

Comment:
If the function returned a *v1beta1.Task, all of the returns for the error cases can become `return nil, fmt.Errorf("....")`

Reply:
I see. Then I need to change all my functions return a pointer instead of a value.